File: Filas/tickets.js
function timeRequiredToBuy(tickets, k) {

    let queue = [];
    const size = tickets.length;

    // Cria a fila com os índices para cada pessoa
    for(let i = 0; i < size; i++) {
        queue.push(i);
    }

    let total = 0;

    while(true) {

        let i = queue.shift(); // Retira o índice da pessoa da fila

        tickets[i] -= 1;
        total++;

        // Se a pessoa não comprou tudo, retorna
        if(tickets[i] != 0) {
            queue.push(i);
        } 

        if(tickets[k] === 0) return total;

    }
}

console.log(timeRequiredToBuy([1 ,1], 1));
console.log(timeRequiredToBuy([2, 3, 2], 2));

File: Filas/java/Tickets.java
import java.util.LinkedList;
import java.util.Queue;

public class Tickets {

    public static int timeRequiredToBuy(int[] tickets, int k) {

        int total = 0; int size = tickets.length;
        Queue<Integer> queue = new LinkedList<>();

        for(int i = 0; i < size; i++) {
            queue.add(i);
        }

        while(true) {

            int i = queue.poll();
            tickets[i]--;
            total++;

            if(tickets[i] != 0) {
                queue.add(i);
            }

            if(tickets[k] == 0) return total;
        }
        
    }
    public static void main (String[] args) {

        int[] tickets = {2, 3, 2};
        System.out.println(timeRequiredToBuy(tickets, 2));
    }
}